### ISSUE TITLE
InvalidOperationException: Collection was modified; enumeration operation may not execute.

### ISSUE THREAD

**Issue by Red-Cat-Fat** [Rainboom7/TowerDefense#3]
Ошибка в самой игре. 
Логи:
> InvalidOperationException: Collection was modified; enumeration operation may not execute.
> System.ThrowHelper.ThrowInvalidOperationException (System.ExceptionResource resource) (at <437ba245d8404784b9fbab9b439ac908>:0)
> System.Collections.Generic.List`1+Enumerator[T].MoveNextRare () (at <437ba245d8404784b9fbab9b439ac908>:0)
> System.Collections.Generic.List`1+Enumerator[T].MoveNext () (at <437ba245d8404784b9fbab9b439ac908>:0)
> PoisonTower.Shoot () (at Assets/Scripts/Objects/Towers/PoisonTower.cs:39)
> PoisonTower.FixedUpdate () (at Assets/Scripts/Objects/Towers/PoisonTower.cs:18)

**Comment by Rainboom7** [comment]
https://github.com/Rainboom7/TowerDefense/commit/ffd16522c2bf4ac1b80a2a73be2b6dce5f70352d
Не знаю насколько это должно работать для юнити, но попытка не пытка

**Comment by Red-Cat-Fat** [comment]
Нет. Проблема не в этом. Разберись из-за чего вообще возникает такая ошибка, а потом уже иди её исправлять

**Comment by Rainboom7** [comment]
ну так ошибка как я понял в модификации коллекции в ontriggerexit в то время как foreach итерирует ее для нанесения урона, тут по факту должна помочь потокобезопасная коллекция, ну или вынесение получения урона в метод самого врага, что наверное не очень красиво будет выглядеть
